enhancements/olm/operator-inspect.md

```diff
@@ -0,0 +1,131 @@
+  - "@ecordell"
+  - "@njhale"
+approvers:
+  - "@ecordell"
+  - "@njhale"
+creation-date: 2020-1-6
+last-updated: 2020-1-6
+status: implementable
+see-also:
+  - "./enhancements/must-gather.md"
+  - "./enhancements/oc/inspect.md"
+---
+
+# Operator Must Gather
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Open Questions [optional]
+
+1. This enhancement defines a library, which will be used to create commands in multiple other projects. I think that initially the `operator-sdk` and `opm` tools are the correct fit, but eventually this could be apart of the `oc adm inspect` command.
+2. Should this enhancement be made in operator-framework as well.
+
+## Summary
+
+Inspecting the cluster has been a popular command that was added to the `oc` command. This command helps admins, support, and developers gather information for the controlers and resources that they are managing/writing or debugging. Operators deployed by OLM have their own set of resources that must be captured for easy debugging. These resources may be spread accross multiple namespaces, may include many different resources such as RBAC rules, and OLM specific resources. This means we need to teach `oc adm inspect` to understand OLM resources, such as subscription.
+
+## Motivation
+
+Similar motivation for the [must gather]("./enhancements/must-gather.md#motivation") enhancement. This will be a tool that all types of operators shipping through OLM can use during support of their operator. Similar principles to the `must-gather` will be followed such as `It must be simple` and `this tool is about the first shotgun gathering so you only have to ask the customer once.`. 
+
+
+### Goals
+
+List the specific goals of the proposal. How will we know that this has succeeded?
+1. Gathering should gather exactly the operator and the resources created by OLM.
+2. A library that can be used to enable this functionality across multiple commands.
+3. A small set of user facing CLI flags. 
+4. In a failing cluster, gathering should be maximized to collect everything it can even when part of it fails.
+5. Should be 100% functional when not in a openshift environment.
+
+### Non-Goals
+
+1. Anything related to the wider cluster. This is not meant to be a OLM debug tool.
+
+## Proposal
+
+A library, in the [api](https://github.com/operator-framework/api/tree/master/pkg) repo that will be vendored by `opm` and `operator-sdk`. This library will include all of the traversal logic for retrieving OLM Operators data. 
+
+The library will include a set of pflags that can be used by commands to create the options structure.
+
+### Implementation Details/Notes/Constraints [optional]
+
+#### Constraints
```

Comment:
> `this tool is about the first shotgun gathering so you only have to ask the customer once.

we want to gather everything. Just the CSV misses all the things that the install plan created (RBAC, Prometheus etc). 

This may be useful to implement in the library in the future, but I think it should be left off the for now.